log4j-core/src/main/java/org/apache/logging/log4j/core/util/Log4jThreadFactory.java

```diff
@@ -94,4 +109,8 @@ public Thread newThread(final Runnable runnable) {
         return thread;
     }
 
+    private static int getThreadPriority() {
+        int result = PropertiesUtil.getProperties().getIntegerProperty(THREAD_PRIORITY, Thread.NORM_PRIORITY);
+        return Math.min(Math.max(Thread.MIN_PRIORITY, result), Thread.MAX_PRIORITY);
```

Comment:
I would do the following instead:

```suggestion
        final String property = "log4j2.thread.priority";
        int priority = PropertiesUtil.getProperties().getIntegerProperty(property, Thread.NORM_PRIORITY);
        if (priority < Thread.MIN_PRIORITY || priority > Thread.MAX_PRIORITY) {
            StatusLogger.getLogger().warn("invalid {}: {}", property, priority);
            priority = Thread.NORM_PRIORITY;
        }
        return priority;
```

There are many other places in Log4j where threads are created. We might want to choose another name for the property.